File: src/rider/main/kotlin/com/jetbrains/rider/plugins/testlinker/GotoLinkedTypesWithDerivedNameAction.kt
package com.jetbrains.rider.plugins.testlinker

import com.jetbrains.rider.actions.base.RiderAnAction
import icons.ReSharperIcons.UnitTesting

class GotoLinkedTypesWithDerivedNameAction : RiderAnAction(
        "GotoLinkedTypesWithDerivedNameAction",
        "Goto Linked Types With Derived Name",
        null,
        UnitTesting.TestFixtureToolWindow)

File: src/dotnet/ReSharperPlugin.TestLinker/Actions/GotoLinkedTypesWithDerivedNameAction.cs
using JetBrains.Application.DataContext;
using JetBrains.Application.UI.ActionsRevised.Menu;
using JetBrains.Application.UI.ActionSystem.ActionsRevised.Menu;
using JetBrains.ReSharper.Feature.Services.Actions;
using JetBrains.ReSharper.Feature.Services.Navigation.ContextNavigation;
using ReSharperPlugin.TestLinker.Navigation;

namespace ReSharperPlugin.TestLinker.Actions
{
    [Action(
        ActionId: Id,
        Text: "Goto Linked Types With Derived Name",
        Id = 17023,
        IdeaShortcuts = new[] {"Shift+Control+O"},
        VsShortcuts = new[] {"Shift+Control+O"})]
    public class GotoLinkedTypesWithDerivedNameAction
        : ContextNavigationActionBase<LinkedTypesWithDerivedNameNavigationProvider>
    {
        public const string Id = nameof(GotoLinkedTypesWithDerivedNameAction);

        public override IActionRequirement GetRequirement(IDataContext dataContext)
        {
            return CurrentPsiFileRequirementNoCaches.FromDataContext(dataContext);
        }
    }
}
